taskcluster/ac_taskgraph/transforms/gradle_optimization.py

```diff
@@ -17,6 +17,10 @@ def extend_optimization_if_one_already_exists(config, tasks):
             skip_unless_changed = optimization["skip-unless-changed"]
 
             component = task["attributes"]["component"]
+            # TODO Remove this special case when ui-test.sh is able to accept "browser-engine-gecko"
```

Comment:
nit: if the special-case is needed, could we put it in the yaml (e.g. have 2 different variables for the ui-test.sh argument and component name, that default to the same thing but can be overridden)?

Reply:
I think we should get rid of the special-case by changing `ui-test.sh` and then removing this if statement. If we create 2 variables in the YAML, it's less obvious that this special case is not here to stay. I'm going to land this code as is. I'm happy to change it if you feel I should.